tests/lib/rules/no-pause-test.js

```diff
@@ -0,0 +1,129 @@
+
+let rule = require('../../../lib/rules/no-pause-test'),
+  RuleTester = require('eslint').RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const TEST_FILE_NAME = 'some-test.js';
+
+let ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('no-pause-test', rule, {
+  valid: [
+    {
+      filename: TEST_FILE_NAME,
+      code: `
+        import setupApplicationTest from 'ember-qunit';
+        import { module, test } from 'qunit';
+
+        module('Acceptance | module', function(hooks) {
+          setupApplicationTest(hooks);
+
+          test('foo', async function(assert) {
+            assert.dom('.foo').hasText('Bar');
+          });
+        });
+      `,
+    },
+    {
+      filename: TEST_FILE_NAME,
+      code: `
+        import setupApplicationTest from 'ember-qunit';
+        import { module, test } from 'qunit';
+        import { pauseTest } from '@ember/test-helpers';
+
+        module('Acceptance | module', function(hooks) {
+          setupApplicationTest(hooks);
+
+          test('foo', async function(assert) {
+            // await pauseTest();
+            assert.dom('.foo').hasText('Bar');
+          });
+        });
+      `,
+    },
+  ],
+
+  invalid: [
+    {
+      filename: TEST_FILE_NAME,
+      code: `
+        import setupApplicationTest from 'ember-qunit';
+        import { module, test } from 'qunit';
+
+        module('Acceptance | module', function(hooks) {
+          setupApplicationTest(hooks);
+
+          test('foo', async function(assert) {
+            await this.pauseTest();
+          });
+        });
+      `,
+      errors: [
+        {
+          message: 'pauseTest() will cause CI failures.',
+          type: 'CallExpression',
+        },
+      ],
+    },
+    {
+      filename: TEST_FILE_NAME,
+      code: `
+        import setupApplicationTest from 'ember-qunit';
+        import { pauseTest } from '@ember/test-helpers';
+        import { module, test } from 'qunit';
+
+        module('Acceptance | module', function(hooks) {
+          setupApplicationTest(hooks);
+
+          test('foo', async function(assert) {
+            await pauseTest();
+          });
+        });
+      `,
+      errors: [
+        {
+          message: 'pauseTest() will cause CI failures.',
+          type: 'CallExpression',
+        },
+      ],
+    },
+    {
+      filename: TEST_FILE_NAME,
+      code: `
+        import setupApplicationTest from 'ember-qunit';
+        import { module, test } from 'qunit';
+
+        module('Acceptance | module', function(hooks) {
+          setupApplicationTest(hooks);
+
+          test('foo', function(assert) {
+            this.pauseTest();
+          });
+        });
+      `,
+      errors: [
```

Comment:
When you copy the test cases from `no-test-and-then`, you'll see that we also need to add `output: null` in each invalid test case.